build/rust/Cargo.lock

```diff
@@ -412,6 +433,12 @@ version = "0.1.5"
 source = "registry+https://github.com/rust-lang/crates.io-index"
 checksum = "245097e9a4535ee1e3e3931fcfcd55a796a44c643e8596ff6566d68f09b87bbc"
 
+[[package]]
+name = "constant_time_eq"
```

Comment:
There is already a version of `constant_time_eq` in Cargo.lock. Is this version necessary, and if so, can we update the other dependencies so they all use the same version?

Reply:
I believe if we bump `blake2b_simd` to v1.0.1 then we'd drop the old version of `constant_time_eq` and utilize only this newer version. However, from the looks of it bumping to this version now brings in some unsafe rust because they're trying to utilize assembly code to prevent the compiler from optimizing the code in a way that could break the constant time guarantees it looks like.

I'd prefer to stick with `0.1.5` until I've had more time to dig into what [this code](https://github.com/cesarb/constant_time_eq/compare/0.1.5...0.2.5#diff-b1a35a68f14e696205874893c07fd24fdb88882b47c23cc0e0c80a30c7d53759R3-R49) is doing and be sure this is safe.

Reply:
@kdenhartog the issue is not using `constant_time_eq`, it's that we're not using the version we already have in Cargo.lock. We need to avoid having multiple copies of the same dep whenever possible, especially when it brings in transitive deps that have different versions. We need to make the minimum changes to Cargo.lock to satisfy the requirements and not the maximum which is what cargo does by default.